src/components/Containers/FormComponentProvider/FormComponentProvider.tsx

```diff
@@ -0,0 +1,21 @@
+import globalStyle from '@/GlobalClasses.module.scss'
+import { initialValues } from '@/services/initialValues';
+import { SandwichPayload } from '@/services/types';
+import { PropsWithChildren } from 'react';
+import { FormProvider, SubmitHandler, useForm } from 'react-hook-form';
+
+const FormComponentProvider = ({ children }: PropsWithChildren) => {
+  const methods = useForm<SandwichPayload>();
+
+  const onSubmit: SubmitHandler<SandwichPayload> = (data) => {
+    console.log(data);
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <form onSubmit={methods.handleSubmit(onSubmit)} className={globalStyle.formContainer}>{children}</form>
```

Comment:
**nit:** you don't have to nest another `form` and pass these methods from hook. `FormProvider` already has these methods in it. I mean you don't need `useForm` in this case